app/models/game.rb

```diff
@@ -32,7 +32,8 @@ def check(id, opponent_id)
 
   def determine_checkmate
     return checkmate(white_player_id) if turn_number.even? && determine_check
-    checkmate(black_player_id) if turn_number.odd? && determine_check
+    return checkmate(black_player_id) if turn_number.odd? && determine_check
+    false
```

Comment:
It's taking care of the case that the game is not in check. If derermine_check return false this method will not call checkmate at all, right?